src/App.js

```diff
@@ -1,20 +1,441 @@
 import React from "react";
-import logo from "./logo.png";
+
 import "./App.css";
+import Header from "./components/Header";
+
+import TaskList from "./components/TaskList";
+import TaskListInProgress from "./components/TaskListInProgress";
+import TaskListInReview from "./components/TaskListInReview";
+import TaskListCompleted from "./components/TaskListCompleted";
 
 class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      tasks: [
+        {
+          id: 0,
+          title: "Reverse Engineer the Type-33 Needler",
+          task: "Coding a software to reverse engineer the Needler for Sacred Promissory",
+        },
+        // {
+        //   id: 1,
+        //   title: "Fixing bugs",
+        //   task: "Fixing coding bugs on the software for Machine Learning AI",
+        // },
+      ],
+      tasksInProgress: [
+        {
+          // id: 0,
+          // title: "React Website - ORION Project",
+          // task: "Coding a frontend website for Doctor Halsey",
+        },
+      ],
+      tasksInReview: [
+        // {
+        //   id: 0,
+        //   title: "Project Fireteam Osiris",
+        //   task: "Implementing a MongoDB Database for Spartan Locke",
+        // },
+      ],
+      tasksCompleted: [
+        {
+          // id: 0,
+          // title: "Improved M-41 SPNKER with upgrades to tracking",
+          // task: "Added tracking to lock on to targets with greater distance and speed of the payload",
+          // timeTaken: "",
+          // keyTakeaways: "",
+        },
+        // {
+        //   id: 1,
+        //   title: "Fixed software",
+        //   task: "Fixing coding bugs on the software for Machine Learning AI",
+        // },
+      ],
+    };
+  }
+
+  // Transfer tasks from Open Tasks to Tasks In Progress
+  moveTaskOpen = (id) => {
+    // Find the task in state.tasks using the find method
+    const taskToMove = this.state.tasks.find((task) => task.id === id);
+    /** find()
+     * find() method loops through the array elements and returns the first element that is true for task.id === id
+     */
+
+    // if there is a taskToMove
+    if (taskToMove) {
+      const newId = this.state.tasksInProgress.length + 1;
+      const updatedTask = { ...taskToMove, id: newId };
+      /**
+       * const updatedTask = { ...taskToMove, id: newId };
+       * { ...taskToMove }: ... spread operator creates a shallow copy of the taskToMove object and spreads them into a new object
+       * id: newId modifies the id property of the shallow copy and assignes the value of `newId` to the `id` property of the copied object
+       */
+
+      this.setState(
+        (prevState) => ({
+          // Wrap the arrow function in () to represent object literal and not function block!
```

Comment:
this is somewhat wrong. Wrapping in parantheses will return the content therein. Comments describing basic JS syntax are not what we want to use comments for usually.

Reply:
```
=> ({ ... })
```

is the same as 
```
=> {
    return { ... } 
}
```